src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md

```diff
@@ -447,6 +447,135 @@ JSON example:
 }
 ```
 
+## Arterial Spin Labeling perfusion data
+
+Template:
+
+```Text
+sub-<label>/[ses-<label>/]
+    perf/
+       sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_dir-<label>][_run-<index>]_asl.nii[.gz]
+       sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_dir-<label>][_run-<index>]_asl.json
+       sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_dir-<label>][_run-<index>]_aslcontext.tsv
+       [sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_dir-<label>][_run-<index>]_m0scan.nii[.gz]]
+       [sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_dir-<label>][_run-<index>]_m0scan.json]
+       [sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_run-<index>]_labeling.jpg]
+```
+
+The complete ASL timeseries should be stored as a 4D NIfTI file in the original acquisition order, accompanied by two ancillary 
+files: `*_asl.json` and `*_aslcontext.tsv`. The `*_asl.nii.gz` and `*_m0.nii.gz` should contain appropriately scaled data and no scaling factors 
+should be stored in the ancillary files or in the NIfTI header. 
+
+The `m0scan` can either be stored inside the 4D ASL time-series NIfTI file or as a separate NIfTI file, depending on whether it 
+was acquired within the ASL time-series or as a separate scan. These and other M0 options are specified in the required M0 field. 
```

Comment:
```suggestion
was acquired within the ASL time-series or as a separate scan. These and other M0 options are specified in the required `M0` field of the `*_asl.json` file. 
```
The M0 field is in the asl.json file, right?

Reply:
Yes